src/Npgsql/NpgsqlTransaction.cs

```diff
@@ -225,6 +225,8 @@ public void Save(string name)
                 throw new ArgumentException("name can't be empty", nameof(name));
             if (name.Contains(";"))
                 throw new ArgumentException("name can't contain a semicolon");
+            if (!char.IsLetter(name[0]))
+                throw new ArgumentException("name should start with a letter");
```

Comment:
To be consistent with the other error messages, I'd suggest: "name must start with a letter"